### ISSUE TITLE
[MPIR-376] LightweightHttpsWagon is always used

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-project-info-reports-plugin#227]
**[Jan Gerken](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jan-apache-issues)** opened **[MPIR-376](https://issues.apache.org/jira/browse/MPIR-376?redirect=false)** and commented

For all dependencies, one of the plugins needs to download always the `LightweightHttpsWagon` is used. Independent of the wagon provider configuration in the `settings.xml` or via `-Dmaven.wagon.provider.https` and `-Dmaven.wagon.provider.http`.

Starting with Maven 3.0.4 the `httpclient` is the default wagon provider used for http and https connections. Therefore, I added add `httpConfiguration` for all servers I access in my `settings.xml` to reduce the timeout. This works well with all other plugins but for this plugin in version 3.0.0 it generates the following warning for each download:

```java
[WARNING] Could not apply configuration for central to wagon org.apache.maven.wagon.providers.http.LightweightHttpsWagon:Cannot find 'httpConfiguration' in class org.apache.maven.wagon.providers.http.LightweightHttpsWagon
```

I tried to explicitly define the wagon provider to be `httpclient` for each server in my `settings.xml` and I tried to force the client via the command line arguments. None of this changes was successful.

Please make this plugin use the configured wagon provider for each server and use the same default provider as the Maven version.

---

**Affects:** 3.0.0

**Attachments:**
- [build.log](https://issues.apache.org/jira/secure/attachment/12940657/build.log) (_718.48 kB_)
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12940655/pom.xml) (_1.10 kB_)

**Backported to:** [3.0.1](https://github.com/apache/maven-project-info-reports-plugin/milestone/21?closed=1)

**Comment by jira-importer** [comment]
**[Jan Gerken](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jan-apache-issues)** commented

I have added a minimal example. To get the warning, you need Maven >= 3.0.4 and the following configuration in your `settings.xml`:

```xml
<server>
  <id>central</id>
  <configuration>
    <wagonProvider>httpclient</wagonProvider>
    <httpConfiguration>
      <all>
        <connectionTimeout>5000</connectionTimeout>
      </all>
    </httpConfiguration>
  </configuration>
</server>
```

Additionally, you one of the artefacts must not be present in your local repository. The easiest solution is to rename your `~/.m2/repository` folder. Now call

```java
mvn site
```

I have attached the log of my build.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I see the issue, this is caused by adding the Wagon implementation to the POM. [Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), this has been introduced by MPIR-366. Do we really need that runtime dependency? Maven comes with a HTTP transport by default.

[Jan Gerken](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jan-apache-issues), can you build MPIR yourself from Git and remove the dependency on Lightweight Wagon? Does this work for you?

**Comment by jira-importer** [comment]
**[Jan Gerken](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jan-apache-issues)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) I have built MPIR from Git. Removing the dependency on Lightweight Wagon entirely does not work, it is required by the unit tests. Changing the scope from `runtime` to `test` makes the tests succeed and solves this issue. Should I create a PR on GitHub for it?

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Must be verified with the integration tests. I'm not 100% sure if reducing the scope to test will effect the runtime. Maybe `<optional>` is better in this case.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

All ITs pass locally here.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), all ITs pass on Jenkins. I am going to merge this today.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [3e3851d2f0014cec9f0593a0fa37930f30637121](https://gitbox.apache.org/repos/asf?p=maven-project-info-reports-plugin.git;a=commit;h=3e3851d2f0014cec9f0593a0fa37930f30637121).
